src/plugins/console/public/application/models/sense_editor/sense_editor.ts

```diff
@@ -461,27 +462,33 @@ export class SenseEditor {
 
   getRequestsAsCURL = async (elasticsearchBaseUrl: string, range?: Range): Promise<string> => {
     const requests = await this.getRequestsInRange(range, true);
+    const kibanaBaseUrl = window.location.origin;
     const result = _.map(requests, (req) => {
       if (typeof req === 'string') {
         // no request block
         return req;
       }
 
-      const esPath = req.url;
-      const esMethod = req.method;
-      const esData = req.data;
+      let path = req.url;
+      const method = req.method;
+      const data = req.data;
 
       // this is the first url defined in elasticsearch.hosts
-      const url = es.constructESUrl(elasticsearchBaseUrl, esPath);
+      let url = constructUrl(elasticsearchBaseUrl, path);
 
-      let ret = 'curl -X' + esMethod + ' "' + url + '"';
-      if (esData && esData.length) {
+      if (path.includes(KIBANA_API_KEYWORD)) {
+        path = path.replace(KIBANA_API_KEYWORD, '');
+        url = constructUrl(kibanaBaseUrl, path);
+      }
+
+      let ret = `curl -X${method} '${url}' -H 'kbn-xsrf: reporting'`;
```

Comment:
Could you please add a comment explaining why we need to add the header `'kbn-xsrf: reporting'`?